### PR TITLE
Dev acaldas

### PR DESCRIPTION
**Pull request by acaldas**

### CODE REVIEW

File: src/mock_adapter/controllers/bag.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        session_id = 1
+        bag = models.Bag.get(session_id = session_id, map = True)
+        return bag
+
+    @appier.route("/bags/<int:id>/add/<int:product_id>", "POST")
+    def add_to_bag(self, id, product_id):
+        quantity = self.get_field("quantity", cast = int)
+        session_id = 1
+        product = models.Product.get(id = product_id)
+        bag = models.Bag.get(session_id = id)
```
Comment (tsilva) [inline src/mock_adapter/controllers/bag.py]: create method in bag called get_from_session(), this method returns the session bag

File: src/mock_adapter/controllers/category.py [hunk 2]
```diff
@@ -9,12 +9,12 @@ class CategoryController(appier.Controller):
...
-        limit = self.get_field("limit", 5, int)
+        skip = self.get_field("skip", cast = int)
+        limit = self.get_field("limit", cast = int)
         categories = models.Category.find(skip = skip, limit = limit)
         return categories
 
-    @appier.route("/categories/<id>.png", "GET")
+    @appier.route("/categories/<int:id>.png", "GET")
     def image(self, id):
         path = "resources/images/category{}.png".format(id)
```
Comment (tsilva) [inline src/mock_adapter/controllers/category.py]: format like this instead "resources/images/category%d.png" % id (no particular reason, its just the current standard)

File: src/mock_adapter/controllers/category.py [hunk 3]
```diff
@@ -22,9 +22,10 @@ def image(self, id):
         finally: file.close()
         return data
 
-    @appier.route("/categories/<category>/products.json", "GET")
+    @appier.route("/categories/<int:category>/products.json", "GET")
     def list_products(self, category):
         skip = self.get_field("skip", 0, int)
         limit = self.get_field("limit", 25, int)
-        products = models.Product.find(skip = skip, limit = limit)
+        products = models.Product.find(category = category, skip = skip, limit = limit)
```
Comment (tsilva) [inline src/mock_adapter/controllers/category.py]: break like this (but with 4 spaces before each argument, github is not showing this):

products = models.Product.find(
    category = category, 
    skip = skip, 
    limit = limit
)

File: src/mock_adapter/controllers/category.py [hunk 4]
```diff
@@ -22,9 +22,10 @@ def image(self, id):
...
         return data
 
-    @appier.route("/categories/<category>/products.json", "GET")
+    @appier.route("/categories/<int:category>/products.json", "GET")
     def list_products(self, category):
         skip = self.get_field("skip", 0, int)
         limit = self.get_field("limit", 25, int)
-        products = models.Product.find(skip = skip, limit = limit)
+        products = models.Product.find(category = category, skip = skip, limit = limit)
+        print products
```
Comment (tsilva) [inline src/mock_adapter/controllers/category.py]: dont leave print statements in the code

File: src/mock_adapter/controllers/product.py [hunk 5]
```diff
@@ -3,11 +3,19 @@
...
 import appier
 
+import models
+
 class ProductController(appier.Controller):
 
-    @appier.route("/products/<id>.png", "GET")
+    @appier.route("/products/<int:id>.json", "GET")
+    def show(self, id):
+        product = models.Product.get(id = id, map = True)
```
Comment (tsilva) [inline src/mock_adapter/controllers/product.py]: might as well return it directly in this case: return models.Product.get(id = id, map = True)

File: src/mock_adapter/controllers/product.py [hunk 6]
```diff
@@ -3,11 +3,19 @@
...
+        return product
+
+    @appier.route("/products/<int:id>.png", "GET")
     def image(self, id):
-        file = open("resources/images/bikini1.png", "rb")
+        path = "resources/images/product%s.png" % id
```
Comment (tsilva) [inline src/mock_adapter/controllers/product.py]: %s works, but you know it's an int, so format like this: "resources/images/product%d.png"

File: src/mock_adapter/controllers/user.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+import appier
+import appier_extras
+
+import models
+
+class UserController(appier.Controller):
+
+    @appier.route("/login/", "GET")
+    def login(self):
+        #self.url_for("admin.login_api", username = "root", password = "root")
```
Comment (tsilva) [inline src/mock_adapter/controllers/user.py]: don't leave commented code, if the user controller is no longer being used then delete the controller (and the model as well)

File: src/mock_adapter/main.py [hunk 8]
```diff
@@ -9,6 +9,7 @@ class BaggerApp(appier.WebApp):
     def __init__(self):
         appier.WebApp.__init__(
             self,
+            session_c = appier.session.MemorySession,
```
Comment (tsilva) [inline src/mock_adapter/main.py]: isn't this the default?

File: src/mock_adapter/models/bag.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    )
+
+    session_id = appier.field(
+        type = int
+    )
+
+    def add_product_s(self, product, quantity):
+        if not hasattr(self, "lines"): self.lines = {}
+        product_id = "%s" % product.id
+        _quantity = self.lines.get(product_id, 0)
```
Comment (tsilva) [inline src/mock_adapter/models/bag.py]: can't you use numeric ids in the map instead?

File: src/mock_adapter/controllers/bag.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+import appier
+
+import models
+
+class BagController(appier.Controller):
+
+    @appier.route("/bags/me", "GET")
+    def me_bag(self):
+        bag = models.Bag.get_from_session(map = True)
+        return bag
```
Comment (tsilva) [inline src/mock_adapter/controllers/bag.py]: just one line, so might as well do: return models.Bag.get_from_session(map = True)

File: src/mock_adapter/controllers/bag.py [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+    @appier.route("/bags/<int:id>/add/<int:product_id>", "POST")
+    def add_to_bag(self, id, product_id):
```
Comment (tsilva) [inline src/mock_adapter/controllers/bag.py]: call this method add_product for consistency

File: src/mock_adapter/models/bag.py [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    )
+    
+    @classmethod
+    def get_from_session(cls, **kwargs):
+        session_id = 1
+        return cls.get(session_id = session_id, **kwargs)
+        
+    def add_product_s(self, product, quantity):
+        if not hasattr(self, "lines"): self.lines = {}
+        product_id = "%s" % product.id
```
Comment (tsilva) [inline src/mock_adapter/models/bag.py]: any reason for using string identifiers in the lines map?

File: src/mock_adapter/controllers/bag.py [hunk 13]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+import appier
+
+import models
+
+class BagController(appier.Controller):
+
+    @appier.route("/bags/me", "GET")
```
Comment (tsilva) [inline src/mock_adapter/controllers/bag.py]: call this just /bag

File: src/mock_adapter/controllers/bag.py [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+import models
+
+class BagController(appier.Controller):
+
+    @appier.route("/bags/me", "GET")
+    def me_bag(self):
+        bag = models.Bag.get_from_session(map = True)
+        return bag
+
+    @appier.route("/bags/<int:id>/add/<int:product_id>", "POST")
```
Comment (tsilva) [inline src/mock_adapter/controllers/bag.py]: call this /bag/add (no need for id, get_from_session will be used)

File: src/mock_adapter/models/bag.py (hunk 12 above)
Comment (acaldas) [inline src/mock_adapter/models/bag.py]: Mongo requires that all keys are strings

**Pull request merged**